From 988ddb18e7c0e53019c9c9346932e60d0ce7c688 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 14 Oct 2015 14:01:34 +0530
Subject: [PATCH] [EM/Charger]: Handle error upon charge current throttling

When charge current throttling even if the charger current setting
fails, not returning the error value. So returning the return value
of the charger current throttling.

Change-Id: I614df64f2a3233f5815f660456f44100e89bd5a4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-6770
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Reviewed-on: https://android.intel.com:443/423759
(cherry picked from commit db6121aeb745830047203a75a287d16d377f3973)
---
 drivers/power/power_supply_core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_core.c b/drivers/power/power_supply_core.c
index 5d23976..3331566 100644
--- a/drivers/power/power_supply_core.c
+++ b/drivers/power/power_supply_core.c
@@ -564,9 +564,10 @@ static int ps_set_cur_charge_cntl_limit(struct thermal_cooling_device *tcd,
 	val.intval = state;
 	ret = psy->set_property(psy,
 		POWER_SUPPLY_PROP_CHARGE_CONTROL_LIMIT, &val);
+	if (ret < 0)
+		return ret;
 
-	psy_charger_throttle_charger(psy, state);
-
+	ret = psy_charger_throttle_charger(psy, state);
 	return ret;
 }
 
-- 
1.9.1

